src/Altinn.App.PlatformServices/Implementation/PersonClient.cs

```diff
@@ -0,0 +1,111 @@
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IPersonRetriever"/> that will call the 
+    /// Altinn 3 Register component to retrieve person information.
+    /// </summary>
+    public class PersonClient : IPersonRetriever
+    {
+        private readonly HttpClient _httpClient;
+        private readonly IAppResources _appResources;
+        private readonly IAccessTokenGenerator _accessTokenGenerator;
+        private readonly IUserTokenProvider _userTokenProvider;
+
+        private readonly JsonSerializerOptions _jsonSerializerOptions = new()
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            PropertyNameCaseInsensitive = true
+        };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PersonClient"/> class.
+        /// </summary>
+        /// <param name="httpClient">The HttpClient to be used to send requests to Register.</param>
+        /// <param name="platformSettings">The platform settings from loaded configuration.</param>
+        /// <param name="appResources">A service with access to app specific information.</param>
+        /// <param name="accessTokenGenerator">An access token generator to create an access token.</param>
+        /// <param name="userTokenProvider">A service that can obtain the user JWT token.</param>
+        public PersonClient(
+            HttpClient httpClient, 
+            IOptions<PlatformSettings> platformSettings,
+            IAppResources appResources,
+            IAccessTokenGenerator accessTokenGenerator,
+            IUserTokenProvider userTokenProvider)
+        {
+            _httpClient = httpClient;
+            _httpClient.BaseAddress = new Uri(platformSettings.Value.ApiRegisterEndpoint);
+            _httpClient.DefaultRequestHeaders.Add(
+                General.SubscriptionKeyHeaderName, 
+                platformSettings.Value.SubscriptionKey);
+            _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+            _appResources = appResources;
+            _accessTokenGenerator = accessTokenGenerator;
+            _userTokenProvider = userTokenProvider;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Person?> GetPerson(string nationalIdentityNumber, string lastName, CancellationToken ct)
+        {
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, $"persons");
+
+            AddAuthHeaders(request);
+
+            request.Headers.Add("X-Ai-NationalIdentityNumber", nationalIdentityNumber);
+            request.Headers.Add("X-Ai-LastName", lastName);
+
+            var response = await _httpClient.SendAsync(request, HttpCompletionOption.ResponseContentRead, ct);
+
+            return await ReadResponse(response, ct);
+        }
+
+        private void AddAuthHeaders(HttpRequestMessage request)
+        {
+            Application application = _appResources.GetApplication();
```

Comment:
This is functionality we need for almost all of our clients. Is there a way to consolidate this, with the logic we have [here ](https://github.com/Altinn/app-template-dotnet/blob/main/src/Altinn.App.PlatformServices/Extensions/HttpClientExtension.cs#L21) Or at least make this method more widely available to reuse in our other clients as well?

Reply:
I agree that there is a potential for a more seamless reuse here, but I'm not sure how yet. The code is currently using 3 different services and I don't like sending objects like that around as parameters.

Reply:
I can split the method in two. One method to obtain the values and another one to set the headers. The last one could more easily be reused, but it doesn't change a whole lot. The goal should be to remove all of it and that requires more thought.